src/components/modal/MarkerModal.tsx

```diff
@@ -1,20 +1,49 @@
+// BarChart overview
+// 1. create separate file for the BarChart object & import
+// 2. initialize the 'chartData' array - specifies the data we'll feed to the chart
+// 3. 'data' state formats the chart (sets labels, bars etc.)
+
 import styles from './modal.module.css'
 import { FaHeartbeat } from 'react-icons/fa'
 import { FaDumbbell } from 'react-icons/fa6'
 import { TbStretching } from 'react-icons/tb'
 import Accordion from './Accordion'
+import { useState } from 'react'
+import BarChart from './BarChart'
+import Image from 'next/image'
 
 const iconMappings = {
   physical: <FaDumbbell />,
   cardiovascular: <FaHeartbeat />,
   flexibility: <TbStretching />,
 }
 
-const MarkerModal = ({ openModal, onClose, item, ageKey }: any) => {
-  const params = ageKey.split('_')
-  const first = params[0]
-  const gender = first[0].toUpperCase() + first.slice(1)
-  const age = params.slice(1).join(',').replace(',', '-')
+const MarkerModal = ({ onClose, item }: any) => {
```

Comment:
You will need to add fixed height for this modal to ensure the modal doesn't over run the screen. When I open the flexibility modals the bar chart is expanding the height of the modal too much and I can't reach the `X` to close it.